cockatrice/src/chatview.cpp

```diff
@@ -73,6 +74,14 @@ void ChatView::appendHtml(const QString &html)
         verticalScrollBar()->setValue(verticalScrollBar()->maximum());
 }
 
+void ChatView::appendHtmlServerMessage(const QString &html)
+{
+    bool atBottom = verticalScrollBar()->value() >= verticalScrollBar()->maximum();
+    prepareBlock().insertHtml("<font color=" + SERVER_MESSAGE_COLOR + ">" + html + "</font>");
+    if (atBottom)
+        verticalScrollBar()->setValue(verticalScrollBar()->maximum());
+}
```

Comment:
My reasoning is to give a name to the logic, rather than adding the code wrapped.
This, IMHO, is cleaner and is adding an API instead of manipulating the current one...if that makes sense...